content/05.main-text.md

```diff
@@ -278,41 +278,68 @@ The second uses a Cypher query, prepared by [`hetnetpy.neo4j.construct_dwpc_quer
 Both of these implementations require traversing all paths between the source and target node.
 Hence, they are computationally cumbersome despite optimizations [@doi:10.15363/thinklab.d187].
 
-An alternative approach counts paths by multiplying adjacency matrices.
-However, this approach actually counts walks,
-since it includes sequences of edges that traverse a single node (i.e trail) or edge (i.e. walk) multiple times.
+Since our methods only require weighted counts, not fully enumerated paths, adjacency matrix multiplication presents an alternative approach.
+Multiplication alone, however, counts walks rather than paths, meaning paths traversing a single node multiple times are counted.
 When computing network-based features to quantify the relationship between a source and target node,
 we would like to exclude traversing duplicate nodes (i.e. paths, not trails nor walks) [@doi:10.15363/thinklab.d134].
-Therefore, we invented a suite of algorithms to compute true path counts and DWPCs using matrix multiplication. 
+To benefit from the speed advantages of only counting paths, we developed a suite of algorithms to compute true path counts and DWPCs using matrix multiplication.
 
-TODO: Describe the suite of DWPC algorithms. From the categorize function, there are:
+Our implementation begins by categorizing a metapath according to the pattern of its repeated metanodes, allowing DWPC computation using a specialized order of operations.
+For example, the metapath _DrDtCrC_  is categorized as a set of disjoint repeats, while _DtCtDtC_ is categorized as repeats of the form BABA.
```

Comment:
Would it be helpful to change `_DtCtDtC_` to `_DtCtDpC_` so the example shows that the metanodes and not metaedges are what matters?